### PR TITLE
 KREST-10790 Add test for KafkaRestExceptionMapper.java

### PR DESCRIPTION
**Pull request by msn-tldr**
title

### CODE REVIEW

File: kafka-rest/src/test/java/io/confluent/kafkarest/exceptions/KafkaRestExceptionMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import io.confluent.rest.entities.ErrorMessage;
+import java.util.concurrent.ExecutionException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.errors.TopicDeletionDisabledException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class KafkaRestExceptionMapperTest {
```
Comment (trnguyencflt) [inline kafka-rest/src/test/java/io/confluent/kafkarest/exceptions/KafkaRestExceptionMapperTest.java:31]: I'm a bit surprise that we don't have tests for `KafkaRestExceptionMappe` 😢

File: kafka-rest/src/test/java/io/confluent/kafkarest/exceptions/KafkaRestExceptionMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        40801);
+  }
+
+  private void verifyMapperResponse(Throwable throwable, Status status, int errorCode) {
+    Response response =
+        exceptionMapper.toResponse(throwable);
+    assertNotNull(response);
+    assertEquals(status.getStatusCode(), response.getStatus());
+    ErrorMessage errorMessage = (ErrorMessage) response.getEntity();
+    assertEquals(errorCode, errorMessage.getErrorCode());
```
Comment (trnguyencflt) [inline kafka-rest/src/test/java/io/confluent/kafkarest/exceptions/KafkaRestExceptionMapperTest.java:60]: Could you also assert the message (contains intended string)?

File: kafka-rest/src/main/java/io/confluent/kafkarest/exceptions/KafkaRestExceptionMapper.java [hunk 3]
```diff
@@ -19,7 +19,9 @@
 import io.confluent.rest.entities.ErrorMessage;
 import io.confluent.rest.exceptions.KafkaExceptionMapper;
 import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
 import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.errors.TopicDeletionDisabledException;
 
 public final class KafkaRestExceptionMapper extends KafkaExceptionMapper {
```
Comment (trnguyencflt) [inline kafka-rest/src/main/java/io/confluent/kafkarest/exceptions/KafkaRestExceptionMapper.java:26]: I don't really know the motivation behind `KafkaRestExceptionMapper` vs `KafkaExceptionMapper` but I feel like `TopicDeletionDisabledException` which is a Kafka exception should be mapped in `KafkaExceptionMapper`.

@msn-tldr what do you think?

File: kafka-rest/src/test/java/io/confluent/kafkarest/exceptions/KafkaRestExceptionMapperTest.java (hunk 1 above)
Comment (msn-tldr) [reply kafka-rest/src/test/java/io/confluent/kafkarest/exceptions/KafkaRestExceptionMapperTest.java:31]: Yeah same here, anyways we have a test now.

File: kafka-rest/src/main/java/io/confluent/kafkarest/exceptions/KafkaRestExceptionMapper.java (hunk 3 above)
Comment (msn-tldr) [reply kafka-rest/src/main/java/io/confluent/kafkarest/exceptions/KafkaRestExceptionMapper.java:26]: Good point. I have moved it to `rest-utils`, and left the new test on this PR.

**Review by trnguyencflt — APPROVED** [review]

**Pull request merged**